### PR TITLE
fix: restrict findRoute exposed property

### PR DESCRIPTION
**Pull request by climba03003**
Follow Up #5230 

In the above mentioned PR, it expose the whole store within `find-my-way` which contains not even the schema but the whole `fastify` context which should not be able to modify or see.
We need to decide what should be exposed and minimize it.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]

File: lib/route.js [hunk 1]
```diff
@@ -173,11 +173,23 @@ function buildRouting (options) {
...
+    const route = router.find(
       options.method,
       options.url || '',
       options.constraints
     )
+    if (route) {
+      // we must reduce the expose surface, otherwise
+      // we provide the ability for the user to modify
+      // all the route and server information in runtime
+      return {
```
Comment (Uzlopak) [inline lib/route.js:185]: Should we deepFreeze the Object?

Comment (mcollina) [reply lib/route.js:185]: I generally avoid freezing because it wrecks performance.

Comment (climba03003) [reply lib/route.js:185]: > Should we deepFreeze the Object?

It doesn't matter on this case, `params` and `searchParams` is newly created object based on the current arguments.
`handler` is a function. There is no way to affect the core by this given values.

**Pull request merged**